src/lib/NovaCellAPIClient.ts

```diff
@@ -34,14 +35,19 @@ export type WithCellId<T> = {
   [P in keyof T]: UnwrapAxiosResponseReturn<OmitFirstArg<T[P]>>
 }
 
+// Extend the Configuration type to include axiosInstance
+export interface ExtendedConfiguration extends BaseConfiguration {
+  axiosInstance?: AxiosInstance
```

Comment:
we can just `&` it the way `mock?: boolean` is

if you want to make it a separate type I'd do it like

```
export type NovaCellAPIClientOpts = BaseConfiguration & {
  axiosInstance?: AxiosInstance
  mock?: boolean
}